consensus/serde_utils/src/quoted_u64_fixed_vec.rs

```diff
@@ -0,0 +1,129 @@
+//! Formats `FixedVector<u64,N>` using quotes.
+//!
+//! E.g., `FixedVector::from(vec![0, 1, 2])` serializes as `["0", "1", "2"]`.
+//!
+//! Quotes can be optional during decoding. If `N` does not equal the length of the `Vec`, the `Vec` is truncated.
+
+use serde::ser::SerializeSeq;
+use serde::{Deserializer, Serializer};
+use serde_derive::{Deserialize, Serialize};
+use ssz_types::typenum::Unsigned;
+use ssz_types::FixedVector;
+use std::marker::PhantomData;
+
+#[derive(Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct QuotedIntWrapper {
+    #[serde(with = "crate::quoted_u64")]
+    pub int: u64,
+}
+
+pub struct QuotedIntFixedVecVisitor<N> {
+    _phantom: PhantomData<N>,
+}
+
+impl<'a, N> serde::de::Visitor<'a> for QuotedIntFixedVecVisitor<N>
+where
+    N: Unsigned,
+{
+    type Value = FixedVector<u64, N>;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(formatter, "a list of quoted or unquoted integers")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'a>,
+    {
+        let mut vec = vec![];
+
+        while let Some(val) = seq.next_element()? {
+            let val: QuotedIntWrapper = val;
+            vec.push(val.int);
+        }
+        let fix: FixedVector<u64, N> = FixedVector::from(vec);
```

Comment:
I can see it's subjective, but I think it would be safest if we reject anything that doesn't match the `N` length. This is what we do in `ssz`, so it seems fitting here.

Ideally, we would want to avoid reading any more than `N` elements from `seq` (as to prevent unnecessary allocations).

Reply:
Yes, I think you're totally right. It's even explicit in the API spec: https://github.com/ethereum/eth2.0-APIs/blob/master/types/state.yaml#L71